outwatch/src/main/scala/outwatch/util/Store.scala

```diff
@@ -14,58 +14,63 @@ import scala.util.control.NonFatal
 
 
 object Store {
+  sealed trait BaseAction
+  private final case object Noop extends BaseAction
+  private final case class Error(context: Action, error: Throwable) extends BaseAction
+  trait Action extends BaseAction
 
-  case class Reducer[S, A](reducer: (S, A) => (S, Observable[A]))
+  case class Reducer[A, M](reducer: (M, A) => (M, Observable[A]))
```

Comment:
Should the order of argument here be consistent with the type argument order? Like having `[A,M]` and `(M, A)`.

Thinking about this order change might break existing user code. Does it still make sense to do it?

Reply:
I think thr order here should be the same as the order commonly used in `fold`.

Reply:
@Busti I think this is what @cornerman is trying to say he prefers:

`case class Reducer[M, A](reducer: (M, A) => (M, Observable[A]))`  

I also find this order preferable.  Agree that the fold pattern is ubiquitous.